### PR TITLE
fix: Pdv 3 user registry proteggere le api di lettura con il namespace derivante da api key

### PR DESCRIPTION
**Pull request by BenitoVisone**
- Manage namespace requirement
- Fixed tests for namespace requirement
- Updated CODEOWNERS file

### CODE REVIEW

File: app/src/main/resources/swagger/api-docs.json [hunk 1]
```diff
@@ -30,6 +30,15 @@
             "type" : "string",
             "format" : "uuid"
           }
+        }, {
+          "name" : "namespace",
+          "in" : "query",
+          "description" : "Caller visibility context",
```
Comment (rodoviti) [inline app/src/main/resources/swagger/api-docs.json:36]: i suggest to chang `namespace` description to something like this: `Tenant name`

**Review by rodoviti — APPROVED** [review]

File: core/src/main/java/it/pagopa/pdv/person/core/PersonServiceImpl.java [hunk 2]
```diff
@@ -25,12 +27,12 @@ class PersonServiceImpl implements PersonService {
...
-    public PersonDetailsOperations findById(String id, boolean isNamespaced) {
+    public PersonDetailsOperations findById(String id, Optional<String> namespace) {
         log.trace("[findById] start");
-        log.debug("[findById] inputs: id = {}, isNamespaced = {}", id, isNamespaced);
+        log.debug("[findById] inputs: id = {}, namespace = {}", id, namespace);
         Assert.hasText(id, PERSON_ID_REQUIRED_MESSAGE);
-        if (isNamespaced) {
-            id = findIdByNamespacedId(id);
+        if (namespace.isPresent()) {
+            id = findIdByNamespacedId(id, namespace.toString());
```
Comment (rodoviti) [inline core/src/main/java/it/pagopa/pdv/person/core/PersonServiceImpl.java:35]:
```suggestion
            id = findIdByNamespacedId(id, namespace.get());
```

File: core/src/test/java/it/pagopa/pdv/person/core/PersonServiceImplTest.java [hunk 3]
```diff
@@ -27,14 +27,13 @@ class PersonServiceImplTest {
     @Mock
     private PersonConnector personConnector;
 
-
     @Test
     void findById_nullId() {
```
Comment (rodoviti) [inline core/src/test/java/it/pagopa/pdv/person/core/PersonServiceImplTest.java:31]: add a test case `findById_nullNamespace` like this one

**Review by rodoviti — CHANGES_REQUESTED** [review]

File: core/src/main/java/it/pagopa/pdv/person/core/PersonServiceImpl.java (hunk 2 above)
Comment (BenitoVisone) [reply core/src/main/java/it/pagopa/pdv/person/core/PersonServiceImpl.java]: fixed with https://github.com/pagopa/pdv-ms-person/pull/42/commits/4c55bfe44b3119678e0444a039f44b894afafdc1

File: core/src/test/java/it/pagopa/pdv/person/core/PersonServiceImplTest.java (hunk 3 above)
Comment (BenitoVisone) [reply core/src/test/java/it/pagopa/pdv/person/core/PersonServiceImplTest.java:31]: added with https://github.com/pagopa/pdv-ms-person/pull/42/commits/4c55bfe44b3119678e0444a039f44b894afafdc1

File: app/src/main/resources/swagger/api-docs.json (hunk 1 above)
Comment (rodoviti) [reply app/src/main/resources/swagger/api-docs.json]: fixed with 46283983780a7ffd7b8559d985d1da95749e3bc2

File: web/src/main/java/it/pagopa/pdv/person/web/controller/PersonController.java [hunk 4]
```diff
@@ -43,15 +44,16 @@ public PersonController(PersonService personService) {
             notes = "${swagger.api.person.findById.notes}")
```
Comment (rodoviti) [inline web/src/main/java/it/pagopa/pdv/person/web/controller/PersonController.java:44]: Add a comment to describe the semantic for `id` param based on `namespace` param

Comment (BenitoVisone) [reply web/src/main/java/it/pagopa/pdv/person/web/controller/PersonController.java:44]: added with https://github.com/pagopa/pdv-ms-person/pull/42/commits/6e582c54c7f30a3eb32daf6ad94a088de35171ee

**Review by rodoviti — APPROVED** [review]

**Pull request merged**